jooq-pg/src/main/resources/db/migration/ehr/V15__data_tables_add_parent_num.sql

```diff
@@ -16,137 +16,275 @@
-        UPDATE comp_data_history pa SET num_cap = (
-            select max(ch.num) FROM comp_data_history ch
-            WHERE ch.vo_id=pa.vo_id
-              AND ch.sys_version=pa.sys_version
-              AND pa.entity_idx_len <= ch.entity_idx_len
-              AND ch.entity_idx ^@ pa.entity_idx
-        ) WHERE pa.num_cap = -1;
-        UPDATE ehr_status_data pa SET num_cap = (
-            select max(ch.num) FROM ehr_status_data ch
-            WHERE ch.ehr_id=pa.ehr_id
-              AND pa.entity_idx_len <= ch.entity_idx_len
-              AND ch.entity_idx ^@ pa.entity_idx
-        ) WHERE pa.num_cap = -1;
-        UPDATE ehr_status_data_history pa SET num_cap = (
-            select max(ch.num) FROM ehr_status_data_history ch
-            WHERE ch.ehr_id=pa.ehr_id
-              AND ch.sys_version=pa.sys_version
-              AND pa.entity_idx_len <= ch.entity_idx_len
-              AND ch.entity_idx ^@ pa.entity_idx
-        ) WHERE pa.num_cap = -1;
-        UPDATE ehr_folder_data pa SET num_cap = (
-            select max(ch.num) FROM ehr_folder_data ch
-            WHERE ch.ehr_id=pa.ehr_id
-              AND pa.entity_idx_len <= ch.entity_idx_len
-              AND ch.entity_idx ^@ pa.entity_idx
-        ) WHERE pa.num_cap = -1;
-        UPDATE ehr_folder_data_history pa SET num_cap = (
-            select max(ch.num)
-             FROM ehr_folder_data_history ch
-             WHERE ch.ehr_id=pa.ehr_id
-               AND ch.sys_version=pa.sys_version
-               AND pa.entity_idx_len <= ch.entity_idx_len
-               AND ch.entity_idx ^@ pa.entity_idx
-        ) WHERE pa.num_cap = -1;
-    end if;
-
-end $$;
+-- Create temporary indices for number cap
+CREATE INDEX IF NOT EXISTS mig_ehr_folder_data_history_num_idx ON ehr_folder_data_history (ehr_id) WHERE num_cap = -1 AND num = 0 AND sys_version = 1;
+CREATE INDEX IF NOT EXISTS mig_ehr_folder_data_num_idx ON ehr_folder_data (ehr_id) WHERE num_cap = -1 AND num = 0;
+CREATE INDEX IF NOT EXISTS mig_ehr_status_data_history_num_idx ON ehr_status_data_history (ehr_id) WHERE num_cap = -1 AND num = 0 AND sys_version = 1;
+CREATE INDEX IF NOT EXISTS mig_ehr_status_data_num_idx ON ehr_status_data (ehr_id) WHERE num_cap = -1 AND num = 0;
+CREATE INDEX IF NOT EXISTS mig_comp_data_history_num_idx ON comp_data_history (vo_id) WHERE num_cap = -1 AND num = 0 AND sys_version = 1;
+CREATE INDEX IF NOT EXISTS mig_comp_data_num_idx ON comp_data (vo_id) WHERE num_cap = -1 AND num = 0;
 
-DROP INDEX IF EXISTS mig_ehr_folder_data_history_num_idx;
-DROP INDEX IF EXISTS mig_ehr_folder_data_num_idx;
-DROP INDEX IF EXISTS mig_ehr_status_data_history_num_idx;
-DROP INDEX IF EXISTS mig_ehr_status_data_num_idx;
-DROP INDEX IF EXISTS mig_comp_data_history_num_idx;
-DROP INDEX IF EXISTS mig_comp_data_num_idx;
+--
+-- Migrate
+--
+
+--may be executed in parallel
+CALL ext.mig_num_columns('ehr_folder_data_history'::regclass, 'ehr_id', 'mig_ehr_folder_data_history_num_idx',1000);
```

Comment:
Reduce to maybe 100. Some systems tend to have extensive folder histories.

Reply:
done